.github/workflows/beaver_api.yml

```diff
@@ -0,0 +1,97 @@
+name: Beaver API CI
+
+on:
+  workflow_call:
+env:
+  DEBUG: True
+  SECRET_KEY: ${{ vars.TEST_KEY }}
+  ALLOWED_HOSTS: localhost,127.0.0.1
+  POSTGRES_DB: beaver_api_db
+  POSTGRES_USER: beaver_api
+  POSTGRES_PASSWORD: beaver_api
+  POSTGRES_HOST: 127.0.0.1
+  POSTGRES_PORT: 5432
+  REDIS_HOST: 127.0.0.1
+  REDIS_PORT: 6379
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:15.3
+        env:
+          POSTGRES_USER: ${{ env.POSTGRES_USER }}
+          POSTGRES_PASSWORD: ${{ env.POSTGRES_PASSWORD }}
+          POSTGRES_DB: ${{ env.POSTGRES_DB }}
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd="pg_isready -U beaver_api"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=5
+
+      redis:
+        image: redis:7.2.3
+        ports:
+          - 6379:6379
+        options: >-
+          --health-cmd="redis-cli ping"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=5
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install PostgreSQL and Redis Client Tools
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y postgresql-client redis-tools
```

Comment:
Do we need this?

Reply:
These tools are used to check if services are fully initialised before testing in the step: Wait for Services to be Ready.